system/src/Grav/Common/Page/Medium/ImageMedium.php

```diff
@@ -197,28 +192,19 @@ public function cache()
      */
     public function srcset($reset = true)
     {
-        if (empty($this->alternatives) && empty($this->derivatives)) {
+        if (empty($this->alternatives)) {
             if ($reset) {
                 $this->reset();
             }
+
             return '';
         }
 
-        if (!empty($this->derivatives)) {
-          asort($this->derivatives);
-
-          foreach ($this->derivatives as $url => $width) {
-              $srcset[] = $url . ' ' . $width . 'w';
-          }
-
-          $srcset[] = $this->url($reset) . ' ' . $this->get('width') . 'w';
-        }
-        else {
-          $srcset = [ $this->url($reset) . ' ' . $this->get('width') . 'w' ];
```

Comment:
Previously, calling `$this->url($reset)` here would reset the current medium and remove the registered alternatives. I wasn't sure what the purpose was, but reversing the order so that the `url` method of each alternative is called before the original medium's method did the trick